core/src/commonMain/kotlin/com/saveourtool/processbuilder/ProcessBuilder.kt

```diff
@@ -44,104 +47,88 @@ class ProcessBuilder(private val useInternalRedirections: Boolean, private val f
     )
     fun exec(
         command: String,
-        directory: String,
-        redirectTo: Path?,
-        timeOutMillis: Long,
+        directory: Path? = config.defaultWorkingDirectory,
+        timeOutDuration: Duration = config.defaultExecutionTimeout,
+        redirectTo: ProcessBuilderConfig.Redirects = config.defaultRedirects,
     ): ExecutionResult {
         if (command.isBlank()) {
             logErrorAndThrowProcessBuilderException("Execution command in ProcessBuilder couldn't be empty!")
         }
-        if (command.contains(">") && useInternalRedirections) {
+        if (config.defaultRedirects !is ProcessBuilderConfig.Redirects.None && command.contains(">")) {
             logger.error {
                 "Found user provided redirections in `$command`. " +
-                        "SAVE will create own redirections for internal purpose, please refuse redirects or use corresponding argument [redirectTo]"
+                        "SAVE will create own redirections for internal purpose, " +
+                        "please refuse redirects or use corresponding argument [redirectTo]"
             }
         }
 
         // Temporary directory for stderr and stdout (posix `system()` can't separate streams, so we do it ourselves)
-        val tmpDir = (FileSystem.SYSTEM_TEMPORARY_DIRECTORY /
-                ("ProcessBuilder_" + Clock.System.now().toEpochMilliseconds()).toPath())
-        logger.trace { "Creating temp directory: $tmpDir" }
         // Path to stdout file
-        val stdoutFile = tmpDir / "stdout.txt"
+        val stdoutFile = redirectTo.redirectStdoutTo
         logger.trace { "Creating stdout file of ProcessBuilder: $stdoutFile" }
         // Path to stderr file
-        val stderrFile = tmpDir / "stderr.txt"
+        val stderrFile = redirectTo.redirectStderrTo
         logger.trace { "Creating stderr file of ProcessBuilder: $stderrFile" }
         // Instance, containing platform-dependent realization of command execution
-        val processBuilderInternal = ProcessBuilderInternal(stdoutFile, stderrFile, useInternalRedirections)
-        fs.createDirectories(tmpDir)
-        fs.createFile(stdoutFile)
-        fs.createFile(stderrFile)
-        logger.trace { "Created temp directory $tmpDir for stderr and stdout of ProcessBuilder" }
+        val processBuilderInternal = ProcessBuilderInternal(redirectTo, config.childProcessUsername)
+
+        stdoutFile?.let { path ->
```

Comment:
maybe move this logic to `Redirects` class?

Reply:
Will move creating/deleting logic to `createFiles` and `deleteFiles` methods of `Redirects`